src/components/CurriculumComponents/Banner/Banner.test.tsx

```diff
@@ -0,0 +1,164 @@
+  });
+
+  it("should render the button with correct props", () => {
+    renderBanner();
+
+    const link = screen.getByRole("link");
+
+    expect(link).toHaveAttribute("href", "/teachers/curriculum");
+    expect(link).toHaveAttribute("title", "See curriculum plans");
+    expect(screen.getByText("See curriculum plans")).toBeInTheDocument();
+  });
+
+  it("should render all SVG elements correctly", () => {
+    renderBanner();
+    // Check promotional tag SVG
+    const tagSvg = screen.getByTestId("new-icon").querySelector("svg");
+    expect(tagSvg).toHaveAttribute("aria-hidden", "true");
+    expect(tagSvg).toHaveAttribute("name", "tag-promotional");
+
+    // Check chevron SVG
+    const buttonIcon = screen.getByTestId("button-icon");
+    const chevronSvg = buttonIcon.querySelector('svg[name="chevron-right"]');
+    expect(chevronSvg).toBeInTheDocument();
+    expect(chevronSvg).toHaveAttribute("aria-hidden", "true");
+  });
+
+  it("should render custom newText and ctaText correctly", () => {
+    const customProps = {
+      ...defaultProps,
+      newText: "New feature available",
+      ctaText: "Try it now",
+    };
+
+    renderBanner(customProps);
+
+    expect(screen.getByText("New feature available")).toBeInTheDocument();
+    const link = screen.getByRole("link");
+    expect(screen.getByText("Try it now")).toBeInTheDocument();
+    expect(link).toHaveAttribute("title", "Try it now");
+  });
+
+  it("should support different page types for the CTA link", () => {
+    const pageMap = {
+      careers: "/teachers/careers",
+      classroom: "/teachers/classroom",
+      contact: "/teachers/contact",
+      "curriculum-landing-page": "/teachers/curriculum",
+      "develop-your-curriculum": "/teachers/develop-your-curriculum",
+      home: "/teachers",
+      help: "/teachers/help",
+      labs: "/teachers/labs",
+      "lesson-planning": "/teachers/lesson-planning",
+      "oak-curriculum": "/teachers/oak-curriculum",
+      onboarding: "/teachers/onboarding",
+      "onboarding-role-selection": "/teachers/onboarding/role-selection",
+      "onboarding-school-selection": "/teachers/onboarding/school-selection",
+      "onboarding-use-of-oak": "/teachers/onboarding/use-of-oak",
+      "our-teachers": "/teachers/our-teachers",
+      "support-your-team": "/teachers/support-your-team",
+      "teacher-hub": "/teachers/teacher-hub",
+    } as const;
+
+    Object.keys(pageMap).forEach((pageType) => {
+      const { unmount } = renderBanner({
+        ...defaultProps,
+        page: pageType as SimplePageType,
+        ctaText: `Go to ${pageType}`,
+      });
+
+      const link = screen.getByRole("link");
+      expect(link).toHaveAttribute("href");
+      expect(link.getAttribute("href")).not.toBe("");
+      expect(link).toHaveAttribute("title", `Go to ${pageType}`);
+
+      unmount();
+    });
+  });
+
+  it("should handle OakColor backgrounds", () => {
+    const colors = [
```

Comment:
minor: you can just test a couple of colors

Reply:
Noted